content/platforms/kubernetes/pod-scheduling.md

```diff
@@ -0,0 +1,187 @@
+the ability to describe the selection of nodes for Redis Enterprise cluster
+node pod scheduling.
+
+A Redis Enterprise cluster has a StatefulSet which manages the
+Redis Enterprise cluster node pods. When the cluster is created, the cluster
+is resized, or a pod fails, the scheduler will attempt to choose a node
+to deploy a new Redis Enterprise cluster node pod. You can control this
+scheduling in various ways.
+
+## Using node selectors
+
+The [`nodeSelector`](https://github.com/RedisLabs/redis-enterprise-k8s-docs/blob/master/docs/operator.md#redisenterpriseclusterspec)
+property of the cluster specification uses the same valus and structures as
+[K8s' `nodeSelector`](https://kubernetes.io/docs/concepts/configuration/assign-pod-node/#nodeselector). In general,
+node labels are a simply way to ensure certain nodes will be used for Redis Entperise pods.
+For example, if nodes 'n1' and 'n2' are labeled as "high memory":
+
+```sh
+kubectl label nodes n1 memory=high
+kubectl label nodes n2 memory=high
+```
+
+The Redis Enterprise cluster CRD can request to be scheduled on these nodes:
+
+```yaml
+apiVersion: app.redislabs.com/v1
+kind: RedisEnterpriseCluster
+metadata:
+  name: rec
+spec:
+  size: 3
+  nodeSelector:
+     memory: high
+```
+
+Subsequently, when the operator creates the StatefulSet associated with the pod, the nodeSelector
+section is part of the pod specification. Thus, when the scheduler attempts to
+create new pods, it will need to satisfy the node selection constraints.
+
+
+## Using node pools
+
+A node pool is a typically part of the underlying infrastructure of the K8s cluster
+deployment and provider. Often, node pools are similarly-configured classes of
+nodes (e.g., nodes with the same allocated amount of memory and CPU). Implementors
+will often label these nodes with a consistent set of labels.
+
+On GKE, all node pools have the label `cloud.google.com/gke-nodepool` with a
+value of the name used during configuration. On Azure AKS, node pools can be
+created with a specific set of labels.
+
+As in the previous section, the `nodeSelector` section can be used to request a specific node
+pool by label values. For example, on GKE:
+
+```yaml
+apiVersion: app.redislabs.com/v1
+kind: RedisEnterpriseCluster
+metadata:
+  name: rec
+spec:
+  size: 3
+  nodeSelector:
+     cloud.google.com/gke-nodepool: 'high-memory'
+```
+
+## Using pod anti-affinity
+
+By default, the Redis Enterprise node pods are schedule to avoid being placed
+on the same node for the same cluster:
+
+```yaml
+podAntiAffinity:
+  requiredDuringSchedulingIgnoredDuringExecution:
+  - labelSelector:
+      matchLabels:
+        app: redis-enterprise
+        redis.io/cluster: rec
+        redis.io/role: node
+    topologyKey: kubernetes.io/hostname
+```
+
+Each pod has the three labels above where `redis.io/cluster` is the name of your
+cluster.
+
+You can override this rule to restrict or widen the nodes on which Redis Enterprise
```

Comment:
@alexmilowski are you sure Operator won't just override as it controls the StatefulSet? I would not have customers touch nodeAffinity

Reply:
The StatefulSet for the cluster has a `podAntiAffinity` that is generated. What is specified in the spec is copied verbatim [here in resources.go](https://github.com/RedisLabs/redis-enterprise-operator/blob/ee408af23f757bceca350983864954e2266d018c/pkg/controller/redisenterprisecluster/resources.go#L585). It is never updated unless the CR is updated. There does not appear to be any override except that `podAntiAffinity` and `antiAffinityAdditionalTopologyKeys` cannot both be set.